docs/source/guides/libraries.rst

```diff
@@ -301,12 +272,313 @@ Python Library
+- protobuf
+- python (python-dev)
+
+Install Iroha python libraries:
+"""""""""""""""""""""""""""""""
+
+- Via PIP
+
+  .. code:: sh
+
+      pip install iroha
+
+  For the latest version
+
+  .. code:: sh
+
+      pip install -i https://testpypi.python.org/pypi iroha
+
+- Source code
+
+  .. code:: sh
+
+      git clone https://github.com/hyperledger/iroha.git
+      cd iroha
+
+  For the latest version checkout to develop branch by adding *-b develop* parameter.
+
+  .. code:: sh
+
+      cmake -H. -Bbuild -DSWIG_PYTHON=ON -DSHARED_MODEL_DISABLE_COMPATIBILITY=ON -DSUPPORT_PYTHON2=ON;
+      cmake --build build --target irohapy -- -j4
+
+      - SWIG_PYTHON=ON forces to build bindings for Python.
+      - SHARED_MODEL_DISABLE_COMPATIBILITY=ON disables backward compatibility with old model of Iroha. Since you want to build only client library you don't need to have the compatibility.
+      - SUPPORT_PYTHON2=ON shows that bindings will be built for Python 2. For Python 3 skip this parameter.
+
+  After this you can find Iroha python library in **iroha/build/shared_model/bindings** folder, where you have previously cloned repository.
+
+Compile protobuf mudules of Iroha from schema files:
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Iroha communicates with users through protobuf messages. In order to send transactions and queries to Iroha node you need to get python module for generating protobuf messages.
+First of all you need to clone schema folder of Iroha repository. If you have already cloned Iroha repository in the previous step, just use schema folder from there.
+
+
+Prerequisites
+"""""""""""""
+
+- protobuf
+- pip
+
+  .. code:: sh
+
+      pip install grpcio_tools
+      mkdir iroha-schema
+      git -C iroha-schema init
+      git -C iroha-schema remote add -f schema https://github.com/hyperledger/iroha.git
+      git -C iroha-schema config core.sparseCheckout true
+      echo "schema" >> iroha-schema/.git/info/sparse-checkout
+      git -C iroha-schema pull schema develop
+      cd iroha-schema
+      protoc --proto_path=schema --python_out=. block.proto primitive.proto commands.proto queries.proto responses.proto endpoint.proto
+      python -m grpc_tools.protoc --proto_path=schema --python_out=. --grpc_python_out=. endpoint.proto yac.proto ordering.proto loader.proto
+
+  Protobuf files can be found in **iroha-schema** folder ('\*_pb2\*.py' files)
+
 How to Use/Import
 ^^^^^^^^^^^^^^^^^
 
+In order to specify Iroha libraries location:
+
+.. code:: sh
+
+  import sys
+  sys.path.insert(0, 'path/to/iroha/libs')
+
+
+Import Iroha and all of the protobuf modules that you need:
+
+.. code:: sh
+
+  import iroha
+  import block_pb2
+  import endpoint_pb2
+  import endpoint_pb2_grpc
+  import queries_pb2
+
 Example Code
 ^^^^^^^^^^^^
 
+.. Note::
+
+    Work with byte arrays is different in Python 2 and Python 3. Due to this fact, the work with hashes and blobs is different in the examples.
```

Comment:
There is the note, but it is not anyhow highlighted which examples have to run in Python 2 and which in Python 3.

Please consider adding either shebang line as the first line of examples or any other marker.
Shebang lines may look like those: ``#!/usr/bin/env python3`` - for Python 3 listings and ``#!/usr/bin/env python2`` or just ``#!/usr/bin/env python`` for Python 2.